examples/1-with-urql-client/pages/index.tsx

```diff
@@ -2,25 +2,26 @@ import React from 'react';
 import Head from 'next/head';
 import { withUrqlClient } from 'next-urql';
 
-import PokemonList from '../components/pokemon_list';
+import { NextContext } from '../../../node_modules/@types/next/index';
+import PokémonList from '../components/pokemon_list';
 
-const Home = () => (
+const Home: React.FC = () => (
   <div>
     <Head>
       <title>Home</title>
       <link rel="icon" href="/static/favicon.ico" />
     </Head>
 
-    <PokemonList />
+    <PokémonList />
   </div>
 );
 
-export default withUrqlClient(ctx => {
+export default withUrqlClient((ctx: NextContext) => {
```

Comment:
So the type of the context object we use is _slightly_ different than the one Next exposes, because we actually extend it by shoving the `urqlClient` instance onto it. There's also (at least in the v8 types) no support for `AppTree`, which we need in order to run `ssrPrepass`. All of this is to say that we actually want to use a _different_ type here:

```typescript
import { withUrqlClient, NextContextWithAppTree } from 'next-urql';

...

export default withUrqlClient((ctx: NextContextWithAppTree) => {
```

This does make me think we should both upgrade our types _and_ rename this type to something like `NextContextWithUrql` or `NextUrqlContext` or something of that nature. For this PR, just swap out the type we're using as shown in the diff above.

Reply:
Ahh I see. Could we also expose the `NextContext` that `next-urql` uses? It looks like `withUrqlClient` takes a `NextUrqlClientConfig`, which looks like this:

```ts
type NextUrqlClientConfig =
  | NextUrqlClientOptions
  | ((ctx: NextContext<any, any>) => NextUrqlClientOptions);
```

Given this, correct me if I'm wrong, I think it'd be more exact to use `NextContext` over our extended `NextContext` type.

Agreed on upgrading our types! Also! It looks like `NextPageContext` in Next 9 does have [support for `AppTree`](https://github.com/zeit/next.js/blob/canary/packages/next/next-server/lib/utils.ts#L115)! Lmk if I'm looking at it wrong or something 😄

Reply:
Ah I see what you mean. I think the answer might be two fold. First, we should get the root types upgraded, which I think can just mean removing the `@types/next` dependency entirely (since it's been deprecated) and adding `next` as a `devDependency` (since we'll want to use its type information, but nothing else).

To your second point, I think we'll have to rewrite our own `NextUrqlContext` object to include the `urqlClient`, i.e.:

```typescript
interface NextUrqlContext extends NextPageContext {
  urqlClient: Client;
}
```

since we now support accessing the `urqlClient` on the `ctx` object. And then `NextUrqlClientConfig` should have its type updated to:

```typescript
type NextUrqlClientConfig =
  | NextUrqlClientOptions
  | ((ctx: NextUrqlContext) => NextUrqlClientOptions);
```

That way a user can access all of the standard stuff off `ctx` if they want while also having type safety when accessing `ctx.urqlClient`. Does that make sense? Good find on that one!

Reply:
Awesome! Makes sense to me. I created [this issue](https://github.com/FormidableLabs/next-urql/issues/20) to track this work, and I'd be happy to do it 😄